cwms-data-api/src/main/java/cwms/cda/data/dao/AuthDao.java

```diff
@@ -69,28 +70,31 @@ public class AuthDao extends Dao<DataApiPrincipal>{
     public static final String GET_SINGLE_KEY = "select userid,key_name,created,expires from cwms_20.at_api_keys where UPPER(userid) = UPPER(?) and key_name = ?";
     public static final String ONLY_OWN_KEY_MESSAGE = "You may not create API keys for any user other than your own.";
 
-    private boolean hasCwmsEnvMultiOfficeAuthFix;
-    private String connectionUser;
-    private String defaultOffice;
+    private static boolean hasCwmsEnvMultiOfficeAuthFix = false;
+    private static String connectionUser = null;
+    private static String defaultOffice = null;
```

Comment:
Making these static and conditionally setting them in the constructor weirds me out.  I can't say exactly why but it feels like asking for trouble. Like if the first request made is for an un-checked get end-point and the request includes office_id will it cause the static value to get assigned to whatever the first request provided?   At the very minimum it seems like assigning to AuthDao.defaultOffice shouldn't happen if the set_session_user_direct failed - right?

Reply:
This always uses the office id of the context "e.g. cwms -> hq, spk -> spk" not the context of data.

For the 2nd part I think it just won't get used, but I don't think the additional logic is required just for that state.

Reply:
But the short version is that I'm just trying to avoid creating a new instance of this object on every request since it will be used on a LOT of requests.

Reply:
ok. the part about the context makes more sense.  Should AuthDao have a static getInstance() and just use it singleton style?

Reply:
I considered the use of a singleton, that's going to need more investigation. Since the clientInfo is different for every request i needs the new context on each run so I think for now we do new to just run new on every request since their could be multiple threads. Or synchronize on the object which seems like a terrible idea.

Reply:
Okay, after realizing it needs the javalin Context information of the current request so that setClientInfo is correct for tracing instead of a pure singleton I made it a singleton per thread.